core/io/resource.cpp

```diff
@@ -724,12 +728,13 @@ void Resource::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("emit_changed"), &Resource::emit_changed);
 
 	ClassDB::bind_method(D_METHOD("duplicate", "deep"), &Resource::duplicate, DEFVAL(false));
-	ClassDB::bind_method(D_METHOD("duplicate_deep", "deep_subresources_mode"), &Resource::duplicate_deep, DEFVAL(RESOURCE_DEEP_DUPLICATE_INTERNAL));
+	ClassDB::bind_method(D_METHOD("duplicate_deep", "deep_subresources_mode"), &Resource::_duplicate_deep_bind, DEFVAL(RESOURCE_DEEP_DUPLICATE_INTERNAL));
 
 	// For the bindings, it's much more natural to expose this enum from the Variant realm via Resource.
-	ClassDB::bind_integer_constant(get_class_static(), StringName("ResourceDeepDuplicateMode"), "RESOURCE_DEEP_DUPLICATE_NONE", RESOURCE_DEEP_DUPLICATE_NONE);
-	ClassDB::bind_integer_constant(get_class_static(), StringName("ResourceDeepDuplicateMode"), "RESOURCE_DEEP_DUPLICATE_INTERNAL", RESOURCE_DEEP_DUPLICATE_INTERNAL);
-	ClassDB::bind_integer_constant(get_class_static(), StringName("ResourceDeepDuplicateMode"), "RESOURCE_DEEP_DUPLICATE_ALL", RESOURCE_DEEP_DUPLICATE_ALL);
+	// Therefore, we can't use BIND_ENUM_CONSTANT here because it would
```

Comment:
Missing the end of this sentence.

Wouldn't it make sense to just duplicate (heh) this enum in Resource so it can be exposed normally, and just cast it to the Variant one where relevant?

Reply:
I guess so. On the one hand, this case is already very special (the enum is global in C++ but scoped to `Resource` in C++), so a little bit of special binding code doesn't seem out of place and makes the "bridging" very explicit. Not having to repeat the enum with the `RESOURCE_` prefixes removed is an upside as well. On the other hand, what you suggest is more similar to what is already happening with certain enums in `core_bind.h`. If you ask me, I'd take the opposite direction, that is, integrating those duplicates.

I don't really have a strong opinion beyond personal taste.

In the meantime, I've fixed the sentence.